src/app/services/chat.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import MessageInterface from '../interfaces/chats';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ChatService {
+  private url = 'http://localhost:3000/chat';
```

Comment:
Env var with a sane default please

Reply:
also make it a base url so you can reuse it and just suffix it with /chat or whatever when needed